packages/vite/src/node/publicUtils.ts

```diff
@@ -4,14 +4,19 @@
  * Please control the side-effects by checking the ./dist/node-cjs/publicUtils.cjs bundle
  */
 export { VERSION as version } from './constants'
-export { version as esbuildVersion } from 'esbuild'
```

Comment:
We're already importing `esbuild` from the top-level elsewhere, I think we can use the safer approach and re-export it here then?

Reply:
I think we should keep this, and start importing esbuild dynamically everywhere. I did this here: https://github.com/vitejs/vite/pull/15621/commits/af14ea0d9254709aa5bf77630c640c07d1aaee2a, but if you prefer I can break down this PR in two (one for rollup and one for esbuild).

The first time we need esbuild is for bundling the config file, and hopefully we could find a way to avoid that in some cases in the future. For a .js config file that is self contained (only importing dependendencies, like plugins, etc), we could give a fast path as we don't need to record the imported modules for HMR for example.